src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -318,9 +329,52 @@ private void Process()
                 }
             }
 
+            private void HandleIncomingRequest(SafeCurlMultiHandle multiHandle, IncomingRequest request)
+            {
+                Debug.Assert(!Monitor.IsEntered(_incomingRequests), "Incoming requests lock should only be held while accessing the queue");
+                VerboseTrace("Type: " + request.Type + ", Uri: " + request.Easy.RequestMessage.RequestUri);
+
+                EasyRequest easy = request.Easy;
+                switch (request.Type)
+                {
+                    case IncomingRequestType.New:
+                        ActivateNewRequest(multiHandle, easy);
+                        break;
+
+                    case IncomingRequestType.Cancel:
+                        Debug.Assert(easy._associatedMultiAgent == this, "Should only cancel associated easy requests");
+                        Debug.Assert(easy.CancellationToken.IsCancellationRequested, "Cancellation should have been requested");
+                        FindAndFailActiveRequest(multiHandle, easy, new OperationCanceledException(easy.CancellationToken));
+                        break;
+
+                    case IncomingRequestType.Unpause:
+                        Debug.Assert(easy._associatedMultiAgent == this, "Should only unpause associated easy requests");
+                        if (!easy.EasyHandle.IsClosed)
```

Comment:
Nit: Would be good to consistently use only members or only Properties of EasyRequest